app/Http/Controllers/Auth/DiscordController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Pterodactyl\Http\Controllers\Auth;
+
+use Exception;
+use Pterodactyl\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Container\Container;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Http;
+use Pterodactyl\Http\Controllers\Controller;
+use Pterodactyl\Exceptions\DisplayException;
+use Pterodactyl\Services\Users\UserCreationService;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Contracts\Container\BindingResolutionException;
+
+class DiscordController extends Controller
+{
+    protected AuthManager $auth;
+    protected UserCreationService $creationService;
+
+    /**
+     * @throws BindingResolutionException
+     */
+    public function __construct(UserCreationService $creationService)
+    {
+        $this->auth = Container::getInstance()->make(AuthManager::class);
+        $this->creationService = $creationService;
+    }
+
+    /**
+     * @return JsonResponse
+     */
+    public function login(): JsonResponse
+    {
+        return new JsonResponse(['https://discord.com/api/oauth2/authorize?client_id='.env('DISCORD_CLIENT_ID').'&redirect_uri='.urlencode(env('REDIRECT_URL')).'&response_type=code&scope=identify%20email%20guilds%20guilds.join'], 200, [], null, true);
+    }
+
+    /**
+     * @param Request $request
+     * @return void
+     * @throws DisplayException
+     */
+    public function callback(Request $request)
+    {
+        $code = Http::asForm()->post('https://discord.com/api/oauth2/token', ['client_id' => env('DISCORD_CLIENT_ID'), 'client_secret' => env('DISCORD_CLIENT_SECRET'), 'grant_type' => 'authorization_code', 'code' => $request->input('code'), 'redirect_uri' => env('REDIRECT_URL')]);
```

Comment:
Same thing here - please use multiple lines and indenting where possible. Same goes for any other applicable line below this.